lib/jekyll/readers/post_reader.rb

```diff
@@ -35,7 +35,11 @@ def read_publishable(dir, magic_dir, matcher)
       read_content(dir, magic_dir, matcher).tap do |docs|
         docs.each(&:read)
       end.select do |doc|
-        site.publisher.publish?(doc)
+        publish = site.publisher.publish?(doc)
+        if !publish && site.publisher.hidden_in_the_future?(doc)
+          Jekyll.logger.debug "Skipping:", "#{doc.relative_path} because it is in the future."
+        end
+        publish
```

Comment:
I had no idea you could `#tap` arbitrary values - I've only ever used it on `Enumerable`s before.  I like it, and I've updated the change to use your suggestion.  Thanks!